poly-commitment/src/srs.rs

```diff
@@ -29,6 +30,9 @@ pub struct SRS<G> {
     /// Commitments to Lagrange bases, per domain size
     #[serde(skip)]
     pub lagrange_bases: HashMap<usize, Vec<PolyComm<G>>>,
+
+    #[serde(skip)]
+    lagrange_bases_cache: Option<LagrangeCache<G>>,
```

Comment:
I'm not sure this is the best place to keep it. The only reason I see to keep the cache type as part of the SRS is that we're assuming that this cache might be updated when the SRS is updated.

Our current model is not like that: the SRS can be only updated (with more Lagrange bases) in memory, but this never affects the disk. To update the disk, you need to run a test (see `srs_serialization.rs`). This is optimal in the sense that SRS is only updated when we explicitly want it (and run the test), and not otherwise (which can be very confusing).


I'd probably suggest keeping `LagrangeCache` as a parameter to this test. Like, create a different test that is called something like `populate_lagrange_bases` and if it is run (with a `LagrangeCache` parameter), the test adds/removes bases and updates the cache. 

Am I missing some advantages of the suggested approach?

Reply:
Agree: the cache is something you use, not something that we should carry around in the SRS. Consider a different approach.

Reply:
agreed, is now kept separately